Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -7247,6 +7265,2013 @@ def panorama_install_file_content_update_command(args: dict):
+        summary_data = []
+        for device in topology.all(filter_string=device_filter_str):
+            device.software.check()
+            for version_dict in device.software.versions.values():
+                summary_data.append(dataclass_from_dict(device, version_dict, SoftwareVersion))
+
+        return SoftwareVersionCommandResult(
+            summary_data=summary_data
+        )
+
+    @staticmethod
+    def download_software(topology: Topology, version: str,
+                          sync: bool = False, device_filter_str=None) -> DownloadSoftwareCommandResult:
+        """Download the given software version to the device. This is an async command, and returns
+        immediately."""
+        result = []
+        for device in topology.all(filter_string=device_filter_str):
+            device.software.download(version, sync=sync)
+            result.append(GenericSoftwareStatus(
+                hostid=resolve_host_id(device),
+                started=True
+            ))
+
+        return DownloadSoftwareCommandResult(
+            summary_data=result
+        )
+
+    @staticmethod
+    def install_software(topology: Topology, version: str,
+                         sync: bool = False, device_filter_str=None) -> InstallSoftwareCommandResult:
+
+        result = []
+        for device in topology.all(filter_string=device_filter_str):
+            device.software.install(version, sync=sync)
+            result.append(GenericSoftwareStatus(
+                hostid=resolve_host_id(device),
+                started=True
+            ))
+
+        return InstallSoftwareCommandResult(
+            summary_data=result
+        )
+
+    @staticmethod
+    def reboot(topology: Topology, hostid: str) -> RestartSystemCommandResult:
+
+        result = []
+        for device in topology.all(filter_string=hostid):
+            device.restart()
+            result.append(GenericSoftwareStatus(
+                hostid=resolve_host_id(device),
+                started=True
+            ))
+
+        return RestartSystemCommandResult(
+            summary_data=result
+        )
+
+    @staticmethod
+    def commit(topology: Topology, device_filter_string: str = None) -> List[CommitStatus]:
+
+        result = []
+        for device in topology.active_devices(device_filter_string):
+            job_type = "Commit"
+            job_id = device.commit()
+            if type(device) is Panorama:
+                device_type = "Panorama"
+            else:
+                device_type = "Firewall"
+
+            result.append(CommitStatus(
+                hostid=resolve_host_id(device),
+                job_id=job_id,
+                commit_type=job_type,
+                status="started",
+                device_type=device_type
+            ))
+
+        return result
+
+    @staticmethod
+    def get_commit_job_status(topology: Topology, match_job_ids: List[str] = None) -> List[CommitStatus]:
+        result: List[CommitStatus] = []
```

Comment:
could its name be set to commit_statuses?

Reply:
Easier to name this result, as it's the result of the command. Makes it easier to copy and paste for new functions as well :)